File: src/main/scala/eu/kedev/blog/FlywayConstants.scala
package eu.kedev.blog


object FlywayConstants {

  val prefix = "/*["
  val suffix = "]*/"

  val ids = List("h2", "postgres")

  val driverToId = Map(
    "H2" -> "h2",
    "PostgreSQL" -> "postgres"
  )

  private[blog] def placeholders(chosenId: String) = ids.flatMap {
    case id@`chosenId` => List(s"$id:start" -> "",   s"$id:end" -> "")
    case id            => List(s"$id:start" -> "/*", s"$id:end" -> "*/")
  }.toMap

}

File: src/main/java/eu/kedev/blog/java/FlywayConstants.java
package eu.kedev.blog.java;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static java.util.Collections.unmodifiableList;
import static java.util.Collections.unmodifiableMap;


public class FlywayConstants {

    public static final String prefix = "/*[";
    public static final String suffix = "]*/";

    public static final List<String> ids =
            unmodifiableList(Arrays.asList("h2", "postgres"));

    public static final Map<String, String> driverToId = new HashMap<String, String>() {{
        put("H2", "h2");
        put("PostgreSQL", "postgres");
    }};

    public static Map<String, String> placeholders(final String chosenId) {
        final Map<String, String> out = new HashMap<>();
        for (final String id: ids) {
            if (id.equals(chosenId)) {
                out.put(id + ":start", "");
                out.put(id + ":end", "");
            } else {
                out.put(id + ":start", "/*");
                out.put(id + ":end", "*/");
            }
        }
        return unmodifiableMap(out);
    }
}
